src/components/mapPageComponents/mapView/index.tsx

```diff
@@ -190,29 +192,47 @@ const MapView: React.FC<MapViewProps> = ({
           }
         }
 
-        // Create and add the tree popup to the map
-        const popup = new Popup(treePopupElement);
-        popup.setMap(map);
+        // Creates and adds the tree popup to the map
+        const popup = new Popup(treePopupElement, map);
 
+        // Sets up the autocomplete search bar, only shows places in Boston for suggestions)
         const input = document.getElementById('pac-input') as HTMLInputElement;
-        const autocomplete = new google.maps.places.Autocomplete(input);
-        autocomplete.addListener('place_changed', () => {
-          const place = autocomplete.getPlace();
+        const autocomplete = new google.maps.places.Autocomplete(input, {
+          bounds: BOSTON_BOUNDS,
+          strictBounds: true,
+        });
+        // Services provided by Google Maps
+        const autoService = new google.maps.places.AutocompleteService(); // to retrieve autocomplete predictions
+        const placesService = new google.maps.places.PlacesService(map); // to search for and find details about places
+
+        // A marker to show at the location a user searches for
+        const searchMarker: google.maps.Marker = new google.maps.Marker({
+          map,
+        });
+
+        // Callback function sets searchMarker at place and updates the text in the search input
+        function goToPredictedPlace(
+          place: google.maps.places.PlaceResult,
+          status: google.maps.places.PlacesServiceStatus,
+        ): void {
+          if (status === google.maps.places.PlacesServiceStatus.OK) {
+            goToPlace(place, searchMarker, map, STREET_ZOOM);
+            setSearchInput(place.name);
+          }
+        }
 
+        // Listener for when the user enters a new location
+        autocomplete.addListener('place_changed', () => {
+          const place: google.maps.places.PlaceResult = autocomplete.getPlace();
+          // If the place does not have a geometry (if the user did not enter a valid location)
           if (!place.geometry) {
-            window.alert(
-              "No details available for input: '" + place.name + "'",
-            );
+            // Predicts the place the user wanted and sets the search marker at that place
+            predictPlace(place, autoService, placesService, goToPredictedPlace);
```

Comment:
This is from Google so I assume these predictions are good, did you get a chance to test some incorrect but close addresses/names?

Reply:
Yeah, it works pretty well with finishing names and ignoring typos.